Refactor dev docs, add details about custom builtins (#3574)
DataDog/datadog-agent: docs/dev/checks/builtins/_util.md

# _util

> **This module is intended for internal use and should never be imported directly.**
> Checks should use the methods exposed by the `AgentCheck` class instead, see
> [dedicated docs](https://datadog-checks-base.readthedocs.io/en/latest/) for
> more details.

The module exposes low level functions to run processes from Python integrations.

## Implementation

* [_util.c](/six/common/builtins/_util.c)
* [_util.h](/six/common/builtins/_util.h)
* [util.go](/pkg/collector/python/util.go)

## Functions

```python
def subprocess_output(args, raise_on_empty):
    """Run an external process and return the output.

>     NOTE: If unicode is passed to any of the params accepting it, the

Reviewer: This note is for Python2 only: do we want to update the doc for Python3 later ?

Author: We can change these docs once Python3 becomes the default, in this PR I wanted to keep things easy so that newcomers don't get confused.

>     NOTICE: if unicode is passed to any of the params accepting it, the

Reviewer 2: ```suggestion
    NOTE: If unicode is passed to any of the params accepting it, the
```

>     string will be encoded using the default encoding for the system where the

Reviewer 2: ```suggestion
    string is encoded using the default encoding for the system where the
```

>     Agent is running. If encoding fails, function will raise `UnicodeError`.

Reviewer 2: ```suggestion
    Agent is running. If encoding fails, the function raises `UnicodeError`.
```
